### PR TITLE
Add case export on node

### PR DESCRIPTION
**Pull request by sBouzols**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/test/java/com/powsybl/network/conversion/server/NetworkConversionTest.java [hunk 1]
```diff
@@ -134,6 +138,14 @@ public void test() throws Exception {
...
             assertTrue(mvcResult.getResponse().getContentAsString().startsWith("<?xml version=\"1.0\" encoding=\"UTF-8\"?>"));
 
+            mvcResult = mvc.perform(get("/v1/networks/{networkUuid}/export/{format}", UUID.randomUUID().toString(), "XIIDM").param("variantId", "second_variant_id"))
+                    .andExpect(status().isOk())
+                    .andExpect(content().contentTypeCompatibleWith(MediaType.APPLICATION_OCTET_STREAM))
+                    .andReturn();
+
+            assertEquals("attachment; filename*=UTF-8''20140116_0830_2D4_UX1_pst_second_variant_id.xiidm", mvcResult.getResponse().getHeader("content-disposition"));
+            assertTrue(mvcResult.getResponse().getContentAsString().startsWith("<?xml version=\"1.0\" encoding=\"UTF-8\"?>"));
+
```
Comment (FranckLecuyer) [inline src/test/java/com/powsybl/network/conversion/server/NetworkConversionTest.java:148]: Maybe add a test on non existing variant id ?

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionException.java [hunk 2]
```diff
@@ -59,6 +60,11 @@ public static NetworkConversionException createEquipmentTypeUnknown(String type)
         return new NetworkConversionException(Type.UNKNOWN_EQUIPMENT_TYPE, "The equipment type : " + type + " is unknown");
     }
 
+    public static NetworkConversionException createVariantIdUnknown(String variantId) {
+        Objects.requireNonNull(variantId);
```
Comment (FranckLecuyer) [inline src/main/java/com/powsybl/network/conversion/server/NetworkConversionException.java:64]: useless test : createVariantIdUnknown is always called with non null variantId parameter ...

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java [hunk 3]
```diff
@@ -190,21 +190,32 @@ private Network networksListToMergedNetwork(List<Network> networks) {
...
 
         Exporters.export(format, network, null, memDataSource);
 
         Set<String> listNames = memDataSource.listNames(".*");
         String networkName;
         byte[] networkData;
         if (listNames.size() == 1) {
-            networkName = network.getNameOrId() + listNames.toArray()[0];
+            networkName = network.getNameOrId();
+            if (variantId != null) {
```
Comment (FranckLecuyer) [inline src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java:215]: Maybe always add variant id in the file name ?
If variantId is null, you can use VariantManagerConstants.INITIAL_VARIANT_D

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java (hunk 3 above)
Comment (sBouzols) [reply src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java:215]: Done

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionException.java [hunk 4]
```diff
@@ -20,6 +20,7 @@ public final class NetworkConversionException extends RuntimeException {
     public enum Type {
         UNSUPPORTED_FORMAT(HttpStatus.INTERNAL_SERVER_ERROR),
         UNKNOWN_EQUIPMENT_TYPE(HttpStatus.INTERNAL_SERVER_ERROR),
+        UNKNOWN_VARIANT_ID(HttpStatus.INTERNAL_SERVER_ERROR),
```
Comment (SlimaneAmar) [inline src/main/java/com/powsybl/network/conversion/server/NetworkConversionException.java:23]: BAD_REQUEST ?

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionException.java (hunk 2 above)
Comment (SlimaneAmar) [reply src/main/java/com/powsybl/network/conversion/server/NetworkConversionException.java]: it's a public method maybe better keep it ?

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java [hunk 5]
```diff
@@ -190,24 +190,33 @@ private Network networksListToMergedNetwork(List<Network> networks) {
...
+            }
+        }
 
         Exporters.export(format, network, null, memDataSource);
 
         Set<String> listNames = memDataSource.listNames(".*");
         String networkName;
         byte[] networkData;
+        networkName = network.getNameOrId();
+        networkName += "_" + (variantId == null ? VariantManagerConstants.INITIAL_VARIANT_ID : variantId);
```
Comment (SlimaneAmar) [inline src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java:214]: The user does not know the notion of variant

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java (hunk 3 above)
Comment (SlimaneAmar) [reply src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java]: The user does not know the notion of variant ?

Comment (sBouzols) [reply src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java]: True. Checked with Christine. We should add a user defined name for exported file later.
For now and this scope it's ok.

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionException.java (hunk 4 above)
Comment (sBouzols) [reply src/main/java/com/powsybl/network/conversion/server/NetworkConversionException.java:23]: NOT_FOUND ?

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionException.java (hunk 2 above)
Comment (sBouzols) [reply src/main/java/com/powsybl/network/conversion/server/NetworkConversionException.java:64]: restored

File: src/test/java/com/powsybl/network/conversion/server/NetworkConversionTest.java (hunk 1 above)
Comment (sBouzols) [reply src/test/java/com/powsybl/network/conversion/server/NetworkConversionTest.java:148]: done

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java (hunk 3 above)
Comment (SlimaneAmar) [reply src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java]: ok

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionException.java (hunk 4 above)
Comment (SlimaneAmar) [reply src/main/java/com/powsybl/network/conversion/server/NetworkConversionException.java:23]: Yes

**Review by SlimaneAmar — APPROVED** [review]

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**